### DISCUSSION TITLE
cashRegisterStatusResult

### DISCUSSION THREAD

**Discussion by JancsarZ** [nav-gov-hu/Online-Cash-Register-Logfile#24] [Q&A] [answered]
Sziasztok!

Rendszeresen tervezzük a logfájlok letöltését, a napi könyvelési munka támogatásához.
A QueryCashRegisterStatusRequest szolgáltatás visszaadja a lekérdezett pénztárgépekhez elérhető logfájlok azonosítójának tartományát.

A cashRegisterStatusResult tartalmazza még a lastCommunicationDate és lastFileDate értékeket is, de ezekre szűrni nem lehet a QueryCashRegisterFileDataRequest-ben. Vagyis csak a listában lejött adatokat tudjuk első körben menteni egy saját táblába, ahol azokat a logfájlokat vezetjük, melyeket már letöltöttünk.
APNumber + File_number
Mindössze ebből a két adatból kellene tudnunk, hogy mely logfájlok letöltésére van szükség ill. melyeket dolgoztuk már fel.

Az alábbi kérdések merültek fel bennünk:

A visszakapott logfilehoz tartozó filenumber egy szigorúan növekvő sorszám, ami sosem változik és a végtelenbe tart?
Egy logfile csak egyszer kerülhet feltöltésre, és nem törölhető / módosítható? Vagyis nem változhat meg egy állomány azonosítója?
Ha változhat, akkor a cashRegisterStatusResult adataiból nem tudjuk beazonosítani a már letöltött és letöltendő állományok listáját.
Ha évenként újraindul, akkor szükség lenne egy tárgyévre is, mert nem fogjuk tudni megállapítani hogy melyik intervallumban kell keresni a letöltendő logokat.
Egy tárgyév amúgy is célszerű lenne már a lekérdezés paramétereiben és a válaszban is.
Egy 3 évig használt pénztárgép esetén majd minden válaszban visszakapunk egy ehhez hasonló tartományt: 1-368765 ?
Előre is köszönjük a válaszokat!

**Comment by renced42** [comment, marked as answer]
Kedves @JancsarZ 

1. A visszakapott logfilehoz tartozó filenumber egy szigorúan növekvő sorszám, ami sosem változik és a végtelenbe tart?
Igen, monotonon növekvő sorszám.
2. Egy logfile csak egyszer kerülhet feltöltésre, és nem törölhető / módosítható? Vagyis nem változhat meg egy állomány azonosítója?
Így van, a logfile egyszer szerepel a rendszerben, az nem módosítható. 
3.Ha változhat, akkor a cashRegisterStatusResult adataiból nem tudjuk beazonosítani a már letöltött és letöltendő állományok listáját.
Nem változhat a naplóállomány.
4. Ha évenként újraindul, akkor szükség lenne egy tárgyévre is, mert nem fogjuk tudni megállapítani hogy melyik intervallumban kell keresni a letöltendő logokat.
Nem indul újra.
5. Egy tárgyév amúgy is célszerű lenne már a lekérdezés paramétereiben és a válaszban is.
Nincs jelentősége, mert a naplóállományok jelen pillanatban T-45 nappal visszamenőleg tölthetők le. Tehát nem lehet 3 évre visszamenőleg letölteni azokat.
6. Egy 3 évig használt pénztárgép esetén majd minden válaszban visszakapunk egy ehhez hasonló tartományt: 1-368765 ?
A specifikáció szerint azt a tartományt kapja vissza a kliens ami rendelkezésre áll a letöltéshez vagyis visszakerül a rendelkezésre álló és lekérdezhető első és utolsó állomány sorszáma is a minAvailableFileNumber és maxAvailableFileNumber mezőkben. pl minAvailableFileNumber = 45 , maxAvailableFileNumber = 123. Azaz a 45-123-ig sorszámozott naplóállományokat lehet letölteni.

**Reply by JancsarZ** [reply]
Köszönjük szépen! :)
